webui/react/src/components/TaskTable.tsx

```diff
@@ -1,40 +1,65 @@
-import { Table } from 'antd';
+import { Table, Tooltip } from 'antd';
 import { ColumnsType, ColumnType } from 'antd/lib/table';
 import React, { MouseEventHandler } from 'react';
 
-import { actionsColumn, Renderer, startTimeColumn, stateColumn,
-  userColumn } from 'table/columns';
 import { AnyTask, CommandTask, CommandType, CommonProps } from 'types';
 import { alphanumericSorter } from 'utils/data';
 import { canBeOpened } from 'utils/task';
 import { commandTypeToLabel } from 'utils/types';
 
+import Badge, { BadgeType } from './Badge';
 import Icon from './Icon';
 import { makeClickHandler } from './Link';
 import linkCss from './Link.module.scss';
-import css from './TaskTable.module.scss';
+import {
+  actionsColumn, ellipsisRenderer, Renderer, startTimeColumn, stateColumn, userColumn,
+} from './Table';
+import css from './Table.module.scss';
 
 interface Props extends CommonProps {
   tasks?: CommandTask[];
 }
 
-const typeRenderer: Renderer<CommandTask> = (_, record) =>
-  (<Icon name={record.type.toLowerCase()}
-    title={commandTypeToLabel[record.type as unknown as CommandType]} />);
+const idRenderer: Renderer<CommandTask> = id => {
+  const shortId = id.split('-')[0];
+  return (
+    <Tooltip title={id}>
+      <div className={css.centerVertically}>
```

Comment:
non-blocking: Definitely don't want to bike-shed here, but for this pattern I like [styled-components](https://styled-components.com/), so the style change is local to the code, and to prevent proliferation of CSS files that need to be organized. (For example, it may not be immediately clear that making changes to `Table.module.scss` will affect `TaskTable`. But if this is the pattern we've been using I see no problem here.

Reply:
we started with `styled-components` but ultimately opted to go the `CSS modules` pattern. The few main reasons why:

- style-components completely obfuscate the css classes, making it impossible to query select elements for testing
- CSS-in-JS require additional JS processing for every where it's applied (almost all components) where as CSS modules do not
- CSS-in-JS don't play very nicely with a setup of eslint and TS